src/app/apiTest/hooks/todoQueries.ts

```diff
@@ -1,9 +1,48 @@
-import { useQuery, UseQueryResult } from '@tanstack/react-query';
-import { getAllByUserId, getOneByTodoId } from '@/app/apiTest/calls/todoCalls';
+import { useMutation, UseMutationResult, useQuery, UseQueryResult } from '@tanstack/react-query';
+import {
+  createTodo,
+  deleteTodo,
+  getAllByUserId,
+  getOneByTodoId,
+  updateTodo,
+  updateTodoTime,
+} from '@/app/apiTest/calls/todoCalls';
 import { SelectTodo } from '@/db/schema/todos';
 
+export const useCreateTodo = (): UseMutationResult<
+  SelectTodo,
+  Error,
+  { userId: string; title: string; content: string; startTime: string; endTime: string; color: string }
+> =>
+  useMutation({
+    mutationFn: ({ userId, title, content, startTime, endTime, color }) =>
+      createTodo(userId.toString(), title, content, startTime, endTime, color),
+  });
 export const useGetAllTodos = (userId: string): UseQueryResult<SelectTodo[], Error> =>
   useQuery({ queryKey: ['todos', userId], queryFn: () => getAllByUserId(userId), enabled: !!userId });
 
 export const useGetOneTodo = (todoId: string): UseQueryResult<SelectTodo, Error> =>
   useQuery({ queryKey: ['todo', todoId], queryFn: () => getOneByTodoId(todoId), enabled: !!todoId });
+
+export const useUpdateTodo = (): UseMutationResult<
+  SelectTodo,
+  Error,
+  { todoId: string; title: string; content: string; startTime: string; endTime: string; color: string }
+> =>
+  useMutation({
+    mutationFn: ({ todoId, title, content, startTime, endTime, color }) =>
+      updateTodo(todoId, title, content, startTime, endTime, color),
+  });
```

Comment:
```suggestion
interface UpdateTodoVariables {
  todoId: string;
  title: string;
  content: string;
  startTime: string;
  endTime: string;
  color: string;
}

export const useUpdateTodo = (): UseMutationResult<SelectTodo, Error, UpdateTodoVariables> => {
  const queryClient = useQueryClient();
  return useMutation<SelectTodo, Error, UpdateTodoVariables>({
    mutationFn: ({ todoId, title, content, startTime, endTime, color }) =>
      updateTodo(todoId, title, content, startTime, endTime, color),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['todos'] });
    },
  });
};
```

`queryClient.invalidateQueries` 를 활용한다면 굳이 refetch를 해주지 않아도 됩니다.
이는 mutate가 성공하였을 경우 `todos`로 시작하는 쿼리키의 캐시를 날려 해당 쿼리 요청을 새롭게 해줍니다.
이렇게 된다면 다른 파일에 달려있는 새로고침목적을 위한 refetch를 지울 수 있다고 생각합니다.

Reply:
조언 감사합니다. 적용했습니다 :)